draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -2883,7 +2889,125 @@ https://video.example.com/k/movie1/4/013.ts
+		The prerequisite requirement implies that a previius trigger reaches one of the following states:   
+		<list style="symbols">
+			<t>(i) “complete” or “processed” for successful completion</t>
+			<t>(ii) “failed” for failed processing</t>
+			<t>(iii) “cancelled” for completion of cancellation</t>
+		</list>
+
+
+		</t>
+		<t>
+		Urgency: Separately from priority, uCDN may wish to indicate that dCDN should process a trigger command without delay. This requirement applies to CI/T Trigger commands only,. dCDN MUST process all CI/T Cancel Commands immediately at all times.  Urgency is separate and additional to priority, which indicates order in which trigger commands should be processed yet doesn’t prescribe how soon each command needs to be executed. 
+dCDN MAY reject such requests for urgent processing using “ereject” error code. This can happen due to dCDN internal business logic (e.g. batch-driven purge and invalidation), or due to dependency on other triggers that could not be completed by dCDN immediately. In such a case uCDN should either modify the request removing prerequisites or cancel such pending triggers before re-trying the request.
+		</t>
+
+
+		</list>
+
+	</t>
+	<t>
+	uCDN may combine multiple options in the same trigger command. dCDN should consider the following when processing such commands:
+	<list style="symbols">
+	<t>(i) dCDN MUST reject triggers that are dependent on other pending triggers with lower priority to prevent deadlocks.</t>
+	<t>(ii) When introducing an urgent trigger uCDN SHOULD indicate priority that is equal to or higher than the highest priority among the pending triggers that belong to the same uCDN. dCDN MUST reject urgent triggers that have lower priority than other pending triggers of the same uCDN.</t>
+	<t>(iii) dCDN MUST reject an urgent trigger that has pending triggers as its prerequisite. dCDN MAY reject an urgent trigger that has triggers in “active” or “cancelling” state as its prerequisite, for example when such triggers are deemed to take a long time to complete or to cancel.</t>
+	</list>
+	
+	</t>
+	<t>
+	Object specification:
+
+        <t><list style="empty" hangIndent="3">
+        <t><list style="hanging" hangIndent="3"><t hangText="Name: priority">
+        <vspace blankLines="1"/>
+	Description: Relative weight of the trigger. 
+	When picking a trigger for execution from all pending triggers posted by each uCDN, 
+	the dCDN MUST choose the trigger with the highest priority first. 
+        <vspace blankLines="1"/>
+	Value: Integer from 0 to 100.
+        <vspace blankLines="1"/>
+	Mandatory: No. The value defaults to zero if omitted.
```

Comment:
Thinking about this in terms of unix process priorities... unix allows priority values from -19 to 20.  The default is 0, which allows process priorities to be adjusted up _or_ down with "nice".  With trigger priorities, we're specifying a default of 0, which means that we can only adjust the priority upward.  Would it be more flexible to choose a value in the middle, such as 50?

Reply:
Makes sense, re-defined the range to be from -100 to 100